content/blog/LFX-Mentorship-Blog.md

```diff
@@ -0,0 +1,425 @@
+end;
+
+create script counter;
+
+select event from in into counter;
+select event from counter into in_memory_wal;
+select event from in_memory_wal into out;
+```
+
+**Persistent-wal**
+
+```trickle
+define qos::wal operator on_disk_wal
+with
+  read_count = 20,
+  max_elements = 1000, # Capacity limit of 1000 stored events
+  max_bytes = 10485760 # Capacity limit of 1MB of events
+end;
+create operator on_disk_wal;
+
+define script counter
+script
+  let count = match state==null of
+    case true =>
+      0
+    default =>
+      state.count + 1
+  end;
+
+  let state = {"count": count};
+  {"command": "send_message", "project":"<project-id>", "topic":"<topic-name>", "data": merge event of {"count": state.count} end, "ordering_key": "<ordering-key>"}
+end;
+
+create script counter;
+
+select event from in into counter;
+select event from counter into on_disk_wal;
+select event from on_disk_wal into out;
+```
+
+In the **inbound** trickle, we have the validation logic as follows:
+
+```trickle
+define script validate
+script
+  match state == null of
+    case true =>
+      let valid = match event.data.count == 0 of 
+        case true =>
+          true      
+        default =>
+          false
+      end,
+      let state = {"prev": event.data.count},
+      {"response": event, "valid": valid} 
+      
+    default =>
+      let valid = match state.prev + 1 == event.data.count of 
+        case true =>
+          true
+        default =>
+          false
+      end,
+      let state = {"prev": event.data.count},
+      
+      {"response":event, "valid": valid}
+  end;
+end;
+
+create script validate;
+
+select {"data": event, "meta": $} from in into validate;
+select event from validate into out;
+```
+
+For the validation, we run the source (onramp) and sink (offramp) separately.
+
+## Network Failure Recovery
+
+Testing in poor connectivity to see if guaranteed delivery works
+
+![Network Failure Recovery testing](/img/blog/LFX-Mentorship/network-failure-testing.png)
+
+The pivot point (where it just works) was observed when downlink and uplink packets dropped varies between 47%-50%. 
+
+## Use of Connectors
+
+- Bulk batch rolling event/log storage to GCS
+- Distribution of openTelemetry logs/trace/metrics over GCP pub/sub
+
+## The Tremor community
+
+The Tremor community is absolutely great. As I was contributing to it as a part of my internship, I was lucky to have direct access to the Tremor developers working at Wayfair whenever I had any questions. We used to have a lot of meetings in the General Voice channel on discord with the Tremor developers and anyone could join in and ask questions, discuss ideas and share what they are working on. This used to be super helpful.
+
+Apart from that, every Tuesday we used to have office hours where everyone joins in and there used to be discussions on topics like: "Why did tremor go open source", "Good practices for contributing to open-source", and Q/As. They used to be a lot of fun.
```

Comment:
Office hours are scheduled for the first Tuesday every month not every Tuesday:)

Reply:
whoops :'D Right